WIP Documentation for .NET Core Uninstall Tool (#13562)
dotnet/docs: docs/core/versions/remove-runtime-sdk-versions.md

The parent directories for the SDK and runtime are listed in the output from the `dotnet --list-sdks` and `dotnet --list-runtimes` command, as shown in the earlier table.

---

## .NET Core Uninstall Tool

> The [.NET Core Uninstall Tool (`dotnet-core-uninstall`)](https://dotnet.microsoft.com/download/dotnet-core/uninstall-tool) is a guided tool provided to enable the controlled clean-up of a system such that only the desired versions of .NET Core SDKs and Runtimes remain. A collection of options is available to facilitate the aggregation of uninstallations.

Reviewer: ```suggestion
The [.NET Core Uninstall Tool (`dotnet-core-uninstall`)](https://dotnet.microsoft.com/download/dotnet-core/uninstall-tool) lets you clean-up of .NET Core SDKs and Runtimes on a system such that only the specified versions remain. A collection of options is available to specify which versions are uninstalled. 
```

The documentation of the tool is available [here](../additional-tools/dotnet-core-uninstall.md).

## Visual Studio dependencies on .NET Core SDK versions

> Visual Studio installers prior to Visual Studio 2019 16.3 install a standalone .NET Core SDK, which is listed in the Windows **Add/Remove Programs** dialog. Hence, be aware that removing .NET Core SDKs installed via the Visual Studio installer prior to Visual Studio 2019 16.3 may break the Visual Studio. The following table shows some of the Visual Studio dependencies on .NET Core SDK versions.

Reviewer: ```suggestion
Visual Studio installers prior to Visual Studio 2019 16.3 called the standalone .NET Core SDK installer. As a result, these versions of the SDK appear in the Windows **Add/Remove Programs** dialog. Hence, be aware that removing .NET Core SDKs installed prior to Visual Studio 2019 16.3 may break Visual Studio. If Visual Studio has problems after you uninstall SDKs, run Repair on that version of Visual Studio. The following table shows some of the Visual Studio dependencies on .NET Core SDK versions:
```

Author: "called" or "calls"?

I'm not sure on this.

| Visual Studio version | .NET Core SDK version |
| -- | -- |
| Visual Studio 2019 16.2 (Update 2) | .NET Core SDK 2.2.4xx, 2.1.8xx |
| Visual Studio 2019 16.1 (Update 1) | .NET Core SDK 2.2.3xx, 2.1.7xx |
| Visual Studio 2019 16.0 | .NET Core SDK 2.2.2xx, 2.1.6xx |
| Visual Studio 2017 15.9 (Update 9) | .NET Core SDK 2.2.1xx, 2.1.5xx |

> | Visual Studio 2017 15.8 (Update 8) | .NET Core SDK 2.1.4xx |

Reviewer: Add (@livarcocc please review)

## Removing the NuGet Fallback Folder

Prior to .NET Core SDK 3.0, the .NET Core SDK installers used the `NuGetFallbackFolder` to store a cache of NuGet packages. This cache was used during operations such as `dotnet restore` or `dotnet build /t:Restore`. The `NuGetFallbackFolder` is located at `C:\Program Files\dotnet\sdk` on Windows and at `/usr/local/share/dotnet/sdk` on macOS.

In some cases you may want to remove this folder:

- You are no longer doing .NET development with SDKs below .NET Core SDK 3.0.
- You are doing development on lower SDKs, but you don't mind being on line and things being slower once.

If you want to remove the NuGet Fallback folder, you can delete it, although you will need Admin privileges.

It is generally undesirable to delete the `dotnet` folder. Doing so would remove any global tools you have installed. Also, on Windows, you will break Visual Studio 2019 Update 16.3 (run Repair to recover) and if there are SDK entries in Add/Remove programs they will be orphaned.

Author: Done.

The documentation of the tool is available [here](../additional-tools/dotnet-core-uninstall.md).

## Visual Studio dependencies on .NET Core SDK versions

Visual Studio installers prior to Visual Studio 2019 Update 16.3 call the standalone .NET Core SDK installer. As a result, these versions of the SDK appear in the Windows **Add/Remove Programs** dialog. Hence, be aware that removing .NET Core SDKs installed prior to Visual Studio 2019 Update 16.3 may break Visual Studio. If Visual Studio has problems after you uninstall SDKs, run Repair on that version of Visual Studio. The following table shows some of the Visual Studio dependencies on .NET Core SDK versions:

| Visual Studio version | .NET Core SDK version |
| -- | -- |
| Visual Studio 2019 16.2 (Update 2) | .NET Core SDK 2.2.4xx, 2.1.8xx |
| Visual Studio 2019 16.1 (Update 1) | .NET Core SDK 2.2.3xx, 2.1.7xx |
| Visual Studio 2019 16.0 | .NET Core SDK 2.2.2xx, 2.1.6xx |
| Visual Studio 2017 15.9 (Update 9) | .NET Core SDK 2.2.1xx, 2.1.5xx |
| Visual Studio 2017 15.8 (Update 8) | .NET Core SDK 2.1.4xx |

## Removing the NuGet Fallback Folder

> Prior to .NET Core SDK 3.0, the .NET Core SDK installers used the `NuGetFallbackFolder` to store a cache of NuGet packages. This cache was used during operations such as `dotnet restore` or `dotnet build /t:Restore`. The `NuGetFallbackFolder` is located at `C:\Program Files\dotnet\sdk` on Windows and at `/usr/local/share/dotnet/sdk` on macOS.

Reviewer 2: ```suggestion
Prior to .NET Core SDK 3.0, the .NET Core SDK installers used the `NuGetFallbackFolder` to store a cache of NuGet packages. This cache was used during operations such as `dotnet restore` or `dotnet build /t:Restore`. The `NuGetFallbackFolder` is located at *C:\Program Files\dotnet\sdk* on Windows and at */usr/local/share/dotnet/sdk* on macOS.
```

In some cases you may want to remove this folder:

* You are no longer doing .NET development with SDKs below .NET Core SDK 3.0.
* You are doing development on lower SDKs, but you don't mind being on line and things being slower once.

> If you want to remove the NuGet Fallback folder, you can delete it, although you will need Admin privileges.

Reviewer 2: ```suggestion
If you want to remove the NuGet Fallback folder, you can delete it, although you will need admin privileges.
```

> It is generally undesirable to delete the `dotnet` folder. Doing so would remove any global tools you have installed. Also, on Windows, you will break Visual Studio 2019 Update 16.3 (run **Repair** to recover) and if there are SDK entries in the **Add/Remove Programs** dialog they will be orphaned.

Reviewer 2: ```suggestion
It is generally undesirable to delete the *dotnet* folder. Doing so would remove any global tools you have installed. Also, on Windows, you will break Visual Studio 2019 Update 16.3 (run **Repair** to recover) and if there are SDK entries in the **Add/Remove Programs** dialog they will be orphaned.
```

The parent directories for the SDK and runtime are listed in the output from the `dotnet --list-sdks` and `dotnet --list-runtimes` command, as shown in the earlier table.

---

## .NET Core Uninstall Tool

The [.NET Core Uninstall Tool (`dotnet-core-uninstall`)](https://dotnet.microsoft.com/download/dotnet-core/uninstall-tool) is a guided tool provided to enable the controlled clean-up of a system such that only the desired versions of .NET Core SDKs and Runtimes remain. A collection of options is available to facilitate the aggregation of uninstallations.

> The documentation of the tool is available [here](../additional-tools/dotnet-core-uninstall.md).

Author: Sure, will work on that.

> The documentation of the tool is available [here](../additional-tools/dotnet-core-uninstall.md).

Author: I draft an explanation of the Visual Studio dependency as follows. This section can be added to the end of https://docs.microsoft.com/en-us/dotnet/core/versions/remove-runtime-sdk-versions.

## Visual Studio dependency

Visual Studio installers prior to Visual Studio 2019 16.3 install a standalone .NET Core SDK, which is listed in the Windows **Add/Remove Programs** dialog. Hence, be aware that removing .NET Core SDKs installed via the Visual Studio installer prior to Visual Studio 2019 16.3 may break the Visual Studio. The following table shows the Visual Studio dependencies on the .NET Core SDK versions.

| Visual Studio version | .NET Core SDK version |
| -- | -- |
| Visual Studio 2019 16.2 (Update 2) | .NET Core SDK 2.2.4xx, 2.1.8xx |
| Visual Studio 2019 16.1 (Update 1) | .NET Core SDK 2.2.3xx, 2.1.7xx |
| Visual Studio 2019 16.0 | .NET Core SDK 2.2.2xx, 2.1.6xx |
| Visual Studio 2017 15.9 (Update 9) | .NET Core SDK 2.2.1xx, 2.1.5xx |
| Visual Studio 2017 15.8 (Update 8) | .NET Core SDK 2.1.4xx |

> The documentation of the tool is available [here](../additional-tools/dotnet-core-uninstall.md).

Reviewer: Great! Let's make it "table shows some of the"

We need to check the dependencies with someone on the team. Can you ask Livar or Will who should review?

> The documentation of the tool is available [here](../additional-tools/dotnet-core-uninstall.md).

Author: Done.
